src/graphql/data/nft/Collection.ts

```diff
@@ -0,0 +1,132 @@
+import graphql from 'babel-plugin-relay/macro'
+import { Trait } from 'nft/hooks/useCollectionFilters'
+import { GenieCollection } from 'nft/types'
+import { loadQuery, usePreloadedQuery } from 'react-relay'
+
+import RelayEnvironment from '../RelayEnvironment'
+import { CollectionQuery } from './__generated__/CollectionQuery.graphql'
+
+const collectionQuery = graphql`
+  query CollectionQuery($address: String!) {
+    nftCollections(filter: { addresses: [$address] }) {
+      edges {
+        cursor
+        node {
+          bannerImage {
+            url
+          }
+          collectionId
+          description
+          discordUrl
+          homepageUrl
+          image {
+            url
+          }
+          instagramName
+          isVerified
+          name
+          numAssets
+          twitterName
+          nftContracts {
+            address
+            chain
+            name
+            standard
+            symbol
+            totalSupply
+          }
+          traits {
+            name
+            values
+            stats {
+              name
+              value
+              assets
+              listings
+            }
+          }
+          markets(currencies: ETH) {
+            floorPrice {
+              currency
+              value
+            }
+            owners
+            totalVolume {
+              value
+              currency
+            }
+            listings
+            volume(duration: DAY) {
+              value
+              currency
+            }
+            volumePercentChange(duration: DAY) {
+              value
+              currency
+            }
+            floorPricePercentChange(duration: DAY) {
+              value
+              currency
+            }
+          }
+        }
+      }
+      pageInfo {
+        endCursor
+        hasNextPage
+        hasPreviousPage
+        startCursor
+      }
+    }
+  }
+`
+export function useCollectionQuery(address: string): GenieCollection | undefined {
+  const collectionQueryReference = loadQuery<CollectionQuery>(RelayEnvironment, collectionQuery, {
+    address,
+  })
+  const queryData = usePreloadedQuery<CollectionQuery>(collectionQuery, collectionQueryReference)
+
+  const queryCollection = queryData.nftCollections?.edges[0].node
+  const traits = {} as Record<string, Trait[]>
+  if (queryCollection?.traits) {
+    for (const trait of queryCollection?.traits) {
```

Comment:
nit: prefer mapping/forEach over `for ... of` because it's too close to `for in` and that's unsettling

Reply:
Switched to foreach